doc/design/embargo-mvp.md

```diff
@@ -0,0 +1,131 @@
+Confirming will lock the dandiset for the duration of the release.
+Once the release finishes, the dandiset will be like any other unpublished draft dandiset.
+
+Once created, an embargoed dandiset is only visible or searchable to owners.
+Instead of the `draft` chip, an `embargoed` chip should be used instead.
+
+## CLI
+The CLI experience will be basically unchanged.
+Uploads to an embargoed dandiset will function exactly the same from an API perspective, the data simply goes to a different location.
+
+### TODO: how to browse/download embargoed zarr files?
+Credentials need to be generated, distributed, and used by the client.
+
+
+# Data storage
+Embargoed assets will be stored in a separate S3 bucket.
+This bucket is private and not browseable by the general public.
+
+The API server will use the embargo bucket in exactly the same way it uses the public bucket, but it will only store embargoed data.
+
+When releasing an embargoed dandiset, all asset data for that dandiset is copied to the public bucket.
+
+When uploading a new asset to an embargoed dandiset, the server will first check if that blob has already been uploaded publically.
+If so, the public blob will be used instead of uploading the data again to the embargo bucket.
+
+
+# Django changes
+
+## Models
+The `Dandiset` model will have an `embargo_status` field that is one of `PRIVATE`, `RELEASING`, or `PUBLIC`.
+* `PUBLIC` means that the Dandiset is publically accessible and publishable.
+  This is the state all Dandisets currently have.
+* `PRIVATE` means that the Dandiset is embargoed.
+  It is not searchable or viewable to non-owners.
+  It is not publishable.
+  Manifest YAML/JSON files will be written to the embargo bucket rather than the public bucket.
+* `RELEASING` means that the Dandiset is currently transitioning from embargoed to public.
+  All modification operations will return a 400 error while the release is in progress.
+  This includes metadata changes and uploads.
+
+A new `EmbargoedAssetBlob` model will be added.
+This behaves the same as a normal `AssetBlob`, but points to the embargo bucket rather than the public bucket.
+It also contains a reference to the `Dandiset` it belongs to.
+
+`Asset`s will point to exactly one `AssetBlob` or `EmbargoedAssetBlob`.
+An `Asset` will be considered "embargoed" if it has an `EmbargoedAssetBlob`.
+An embargoed `Asset` can only belong to the dandiset named in the `EmbargoedAssetBlob`.
+The automatically populated `Asset` metadata will also set the `access` field to the appropriate value depending on whether or not the asset is embargoed.
+
+The `Upload` model will have a new optional field `embargoed_dandiset` that points to an embargoed dandiset.
+If specified, the uploaded data will be sent to the embargoed bucket instead of the public bucket, and the `Upload` will create an `EmbargoedAssetBlob` when finalized.
+
+## API
+* Create dandiset endpoint
+
+  Add a new optional `award_number` field.
```

Comment:
we have to be careful with this, since a dandiset can potentially be the result of multiple awards. thus doing it with the metadata is better. i'm not sure moving from a one to one relationship to a one to many relationship is going to be easy. so the alternate thought here is adding an `award` model and then connecting the dandiset to one  or more awards.

Reply:
This is solely so that we can guarantee that an award is specified when creating a new embargoed dandiset. It will be recorded in the metadata. If you had multiple awards, you would pick one to specify initially, then modify the metadata later to include the rest. That might be a subpar UX, but this is the MVP, so I think it's fine for now, and easy to swap out later.

If it's really important to have multiple awards, we could eventually add the ability to specify multiple when creating the dandiset, or we could remove that requirement completely and make users set all of their awards through the meditor.

Reply:
if we are simply using the metadata model instead of adding a field, the multiple award thing works already. the only addition we would add would be in the validator that the embargoed dandiset must have an award associated with it. (ideally we would even check that the nih api returns a valid signal for the award). and the ui could also be updated to simply interface with the nih api. (i would say this isn't essential for MVP). but whether a database model field is created or simply the metadata is used should be decided and my preference would be for the metadata.

Reply:
We agree largely with what you wrote, @satra.

To be explicit: we will design the UI to have a checkbox for creating an embargoed dataset; that will in turn make a required form field for an NIH award number appear. The client will craft a metadata blob that includes the award, and send all this information to the creation endpoint. The creation endpoint will do all the nitty gritty stuff it needs to do thereafter (as described in this document).

If it is imperative to support multiple NIH awards from the jump, it should not be too difficult to design the UI to require one award number, and show a plus button to add more. Then, the "craft a metadata blob" idea from above will simply iterate through a list of award numbers. Otherwise, we can expand to this capability when we implement the full design.

(In the full design, we will also consider the idea of validating an embargoed dandiset--the idea is that an owner could put a valid NIH award number in to get past the initial hurdle, then remove it later in order to have an indefinitely embargoed secret dandiset; by "validating" for qualities such as "does this dandiset have a valid award number", etc., we could simply "freeze" any invalid dandisets, thus preventing use until they become valid. This shouldn't be necessary for the MVP.)

Reply:
just to clarify i don't think it's necessary for the create dandiset ui to have support for multiple awards, just one is sufficient there, and may be a note that additional awards can be added through the metadata editor after the dandiset is created. my comment was more specific to adding a db field that supports only one award.

Reply:
I think there was some confusion in the original language: there was never an intent to track a single award number in the DB, separately from the metadata (but rather, a serializer field in the endpoint for dandiset creation). I think we're all on the same page now.

Reply:
I updated the design of the create dandiset endpoint, it now takes the award information via the metadata as requested.
Resolving